MAUI/Autocomplete/Searching-Filtering.md

```diff
@@ -171,6 +171,276 @@ autocomplete.TextSearchMode = AutocompleteTextSearchMode.Contains;
+
+{% tabs %}
+{% highlight C# %}
+
+/// <summary>
+/// Represents a custom filtering behavior for `Autocomplete` control. 
+/// </summary>
+public class CityFilteringBehavior : IAutocompleteFilterBehavior
+{
+
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+**Step 2:** Then, implement the `GetMatchingIndexes` method of `IAutocompleteFilterBehavior` interface to create your own suggestion list (containing the indices of the filtered items) based on the text entered in the `Autocomplete` control that needs to be shown in the drop-down. The `GetMatchingIndexes` method contains following arguments:
+
+* `source` - The owner of the filter behavior, which holds information about `ItemsSource`  property and so on.
+* `filterInfo` - Contains details about the text entered in `Autocomplete` control. Using this text, you can prepare suggestion list, which gets displayed in the drop-down list. 
+
+The following image shows how to display cities in a drop-down based on the country name entered in the `Autocomplete` control.
+
+{% tabs %}
+{% highlight C# %}
+
+public class CityFilteringBehavior : IAutocompleteFilterBehavior
+{
+    public async Task<object> GetMatchingItemsAsync(SfAutocomplete source, AutocompleteFilterInfo filterInfo)
+    {
+         IEnumerable itemssource = source.ItemsSource as IEnumerable;
+        var filteredItems = (from CityInfo item in itemssource
+                             where item.CountryName.StartsWith(filterInfo.Text, StringComparison.CurrentCultureIgnoreCase) ||
+                                   item.CityName.StartsWith(filterInfo.Text, StringComparison.CurrentCultureIgnoreCase)
+                             select item);
+
+        return await Task.FromResult(filteredItems);
+    }
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+**Step 3:** Applying custom filtering to the `Autocomplete` control by using the `FilterBehavior` property.
+
+{% tabs %}
+{% highlight XAML %}
+ <ContentPage.BindingContext>
+    <local:CityViewModel>
+    </local:CityViewModel>
+</ContentPage.BindingContext>
+  <editors:SfAutocomplete 
+    DisplayMemberPath="CityName"
+    ItemsSource="{Binding Cities}">
+            <editors:SfAutocomplete.FilterBehavior>
+                <local:CityFilteringBehavior/>
+            </editors:SfAutocomplete.FilterBehavior>
+  </editors:SfAutocomplete>
+
+{% endhighlight %}
+{% endtabs %}
+
+The following image demonstrates how to display cities in the drop-down based on the country name entered in the `Autocomplete` control.
+
+![.NET MAUI AutoComplete filter the items based on custom filtering logic.](Images/Searching_Filtering/Custom_filtering.png)
+
+### Choose default item to select
+
+When searching, the first item in the drop-down will be highlighted by default. Using the `SearchBehavior` property, you can customize the default highlighting behavior by using your custom selection logic to select the items based on your search criteria. The default value of `SearchBehavior` is `null`. 
```

Comment:
When searching, the first item in the drop-down will be highlighted by default. Using the `SearchBehavior` property, customize the default highlighting behavior by using your custom selection logic to select the items based on your search criteria. The default value of the `SearchBehavior` is `null.`